### ISSUE TITLE
Direct Stream on Edge Mobile not working

### ISSUE THREAD

**Issue by GitHubby324** [stashapp/stash#4275]
**Describe the bug**
When trying to play a video on Edge Mobile the settings dropdown inside the player is missing the option "direct stream".
While on other browsers (Chrome Mobile, Edge Desktop, Chrome Desktop) "direct stream" is available and streaming works.

**To Reproduce**
1. Open Stash in Edge Mobile
2. Try to play a video
3. Error message pops up for a fraction of a second (too short to read)
4. Loading animation never ends

**Expected behavior**
Should be able to play the video without a problem like on other browsers (Chrome Mobile, Edge Desktop, Chrome Desktop).

**Versions:**
 - Stash: v0.23.1 (build 479e7163)
 - Edge Mobile: 119.0.2088.66
 - Chrome Mobile: 119.0.6045.66
 - Mobile OS: Android 12 (SP1A.210812.016.N975FXXS9HWHA)
 - Device: Samsung Note 10+ (SM-N975F)
 - Stash Host: Raspberry Pi4B (arm64 version of Stash)

**Additional context**
It seems I need "direct stream" since all the other options (webm, dash, hls, ...) do not work on either browser.
I guess this is because I would have to generate transcodes right? (don't want to since I'm running on RPi 4B and are limited resource-wise and like I said - "direct stream" does work with every other browser)

I've already tried many different videos, but none are working on Edge Mobile. (all are MP4 files)

Is this some limitation of Edge Mobile? (altough it's also based on Chromium like Chrome Mobile)

**Comment by Gishan86** [comment]
Problem still persists.

Current Versions:
- Stash: v0.23.1 (build 479e7163)
- Edge Mobile: 120.0.2210.84
- Chrome Mobile: 120.0.6099.144

**Comment by GitHubby324** [comment]
Problem still persists.

Current Versions:
- Stash: v0.24.1 (build bd7d4ac7)
- Edge Mobile: 120.0.2210.84
- Chrome Mobile: 120.0.6099.144

**Comment by GitHubby324** [comment]
Still the same problem.

Current versions:

- Stash: v0.26.0 (build b232e58b)
- Edge Mobile: 125.0.2535.51

**Comment by feederbox826** [comment]
The flashing error is most likely "codec not supported" and the transcoding not working (loading infinitely) sounds like your raspi is too weak to transcode or is transcoding very slowly

While based on chromium, the media playback engine is different and chrome has their own media decoder through VAAPI. HEVC isn't supported which is why the direct stream option isn't available. Desktop has it's own entire decoding stream that can't be compared to mobile

**Comment by Gishan86** [comment]
> The flashing error is most likely "codec not supported" and the transcoding not working (loading infinitely) sounds like your raspi is too weak to transcode or is transcoding very slowly
> 
> While based on chromium, the media playback engine is different and chrome has their own media decoder through VAAPI. HEVC isn't supported which is why the direct stream option isn't available. Desktop has it's own entire decoding stream that can't be compared to mobile

Seems like you misunderstood the issue. It is working on Chrome Mobile but not on Edge Mobile. Both using the same Android phone AND same Raspi.

**Comment by feederbox826** [comment]
Yes, edge mobile has a different media playback system than chrome mobile. Chrome mobile has VAAPI and has tighter integration with the system decoders. While they are both based on chromium, the media decode engine is also in Chrome, not just Chromium

**Comment by Gishan86** [comment]
> Yes, edge mobile has a different media playback system than chrome mobile. Chrome mobile has VAAPI and has tighter integration with the system decoders. While they are both based on chromium, the media decode engine is also in Chrome, not just Chromium

Interesting, didn't know that. It seems to be a browser problem then, thanks for clarifying!
